### PR TITLE
R4R: refactor sync config load way and parser msg handler with route

### PR DESCRIPTION
**Pull request by weichang-bianjie**

### CODE REVIEW

File: handlers/tx.go [hunk 1]
```diff
@@ -13,9 +15,32 @@ import (
...
 	"golang.org/x/net/context"
 	"gopkg.in/mgo.v2/txn"
+	"strings"
 	"time"
 )
 
+var _parser msgparser.MsgParser
+
+func InitRouter(conf *config.Config) {
+	router := msgparser.RegisteRouter()
```
Comment (wanghao-bianjie) [inline handlers/tx.go:25]: RegisteRouter when it is judged that OnlySupportModule is empty.
```suggestion
	var router msgparser.Router
```

File: handlers/tx.go [hunk 2]
```diff
@@ -13,9 +15,32 @@ import (
...
+			if !exist {
+				logger.Fatal("no support module: " + one)
+			}
+			msgRoute = msgRoute.AddRoute(one, fn)
+		}
+		if msgRoute.GetRoutesLen() > 0 {
+			router = msgRoute
+		}
+
+	}
```
Comment (wanghao-bianjie) [inline handlers/tx.go:40]:
```suggestion
	} else {
		router = msgparser.RegisteRouter()
	}
```

File: handlers/tx.go (hunk 1 above)
Comment (wanghao-bianjie) [inline handlers/tx.go:25]: RegisteRouter when it is judged that OnlySupportModule is empty.
```suggestion
	var router msgparser.Router
```

Comment (wanghao-bianjie) [inline handlers/tx.go:25]:
```suggestion
	var router msgparser.Router
```

File: handlers/tx.go (hunk 2 above)
Comment (wanghao-bianjie) [inline handlers/tx.go:40]:
```suggestion
	} else {
		router = msgparser.RegisteRouter()
	}
```

**Review by wanghao-bianjie — COMMENTED** [review]
RegisteRouter when it is judged that OnlySupportModule is empty.

File: handlers/tx.go (hunk 2 above)
Comment (weichang-bianjie) [reply handlers/tx.go:43]: meaningless change

**Review by kaifei-bianjie — APPROVED** [review]

**Pull request merged**